src/System.Memory/src/System/Memory.cs

```diff
@@ -188,7 +188,7 @@ public Span<T> Span
                     // and then cast to a Memory<T>. Such a cast can only be done with unsafe or marshaling code,
                     // in which case that's the dangerous operation performed by the dev, and we're just following
                     // suit here to make it work as best as possible.
-                    return new Span<T>(ref Unsafe.As<char, T>(ref s.GetRawStringData()), s.Length).Slice(_index, _length);
+                    return new Span<T>(Unsafe.As<Pinnable<T>>(s), SpanExtensions.StringAdjustment, s.Length).Slice(_index, _length);
```

Comment:
Using `Unsafe.As<Pinnable<char>>` would be a bit easier to follow, I think.

Reply:
The span constructor expected a Pinnable<T>. If I change it to Pinnable<char>, it fails to find the right constructor:
https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/Span.cs#L117

I don't know why this is happening.
https://ci3.dot.net/job/dotnet_corefx/job/master/job/windows-TGroup_netfx+CGroup_Release+AGroup_x86+TestOuter_false_prtest/5270/console:
`System\Memory.cs(191,40): error CS1503: Argument 1: cannot convert from 'System.Pinnable<char>' to 'T[]' `

Reply:
I see ... leave it as is then.